pytest_splunk_addon_ui_smartx/components/controls/single_select.py

```diff
@@ -355,27 +360,17 @@ def _wait_for_search_list(driver):
             _wait_for_search_list, msg="No values found in SingleSelect search"
         )
 
-    def is_editable(self):
+    def is_editable(self) -> bool:
         """
-        Returns True if the Textbox is editable, False otherwise
+        Returns True if the SingleSelect is editable, False otherwise
         """
         if self.searchable:
-            if self.allow_new_values:
-                self.elements.update(
-                    {
-                        "input": Selector(
-                            select=self.container.select + ' [data-test="textbox"]'
-                        )
-                    }
+            element_key = "combobox_role" if self.allow_new_values else "listbox_role"
+            try:
+                self.get_element(element_key)
```

Comment:
is there any way to check if element is present in DOM at the current moment without waiting?